.github/workflows/ci.yaml

```diff
@@ -46,8 +46,8 @@ jobs:
            && github.repository == 'lampepfl/dotty'
          )"
     steps:
-      - name: Set JDK 16 as default
-        run: echo "/usr/lib/jvm/java-16-openjdk-amd64/bin" >> $GITHUB_PATH
+      - name: Set JDK 17 as default
+        run: echo "/usr/lib/jvm/java-17-openjdk-amd64/bin" >> $GITHUB_PATH
```

Comment:
Merely updating the path here is not sufficient to use JDK 17, since the custom Docker image used by the self-hosted CI runners only contains JDK 16 and JDK 8 (the default) at the moment.

It can be seen in the logs that JDK 8 is actually what was used for this PR run:

https://github.com/lampepfl/dotty/pull/13629/checks?check_run_id=3739562544#step:9:7

```
Run ./project/scripts/sbt ";scala3-bootstrapped/compile; scala3-bootstrapped/test;sjsSandbox/run;sjsSandbox/test;sjsJUnitTests/test;sjsCompilerTests/test ;sbt-test/scripted scala2-compat/* ;configureIDE ;stdlib-bootstrapped/test:run ;stdlib-bootstrapped-tasty-tests/test; scala3-compiler-bootstrapped/scala3CompilerCoursierTest:test"
[info] welcome to sbt 1.5.5 (Private Build Java 1.8.0_282)
```

The repo for the CI Docker image is https://github.com/lampepfl/dotty-ci